Entity/base_game_entity.h

```diff
@@ -54,7 +58,10 @@ class BaseGameEntity {
     [[nodiscard]] int ID() const {
         return m_ID;
     }
+
+    [[nodiscard]] glm::vec3 GetPosition() const {
```

Comment:
I actually discourage this. `m_Position`is something that needs to get read/written. If there is no other logic to be done we don't need Setters/Getters and it just makes not much sense other than satisfying hardcore OOP folks ;)

Reply:
so i just make it public? works for me as well :)